common/src/main/java/org/apache/falcon/entity/DatasourceHelper.java

```diff
@@ -148,16 +152,168 @@ public static String buildJceksProviderPath(URI credURI) {
+                    return isSameString(oCred.getPasswordText(), nCred.getPasswordText());
+                } else if (oCred.getType() == Credentialtype.PASSWORD_FILE) {
+                    return isSameString(oCred.getPasswordFile(), nCred.getPasswordFile());
+                } else if (oCred.getType() == Credentialtype.PASSWORD_ALIAS) {
+                    return (isSameString(oCred.getPasswordAlias().getAlias(),
+                            nCred.getPasswordAlias().getAlias())
+                            && isSameString(oCred.getPasswordAlias().getProviderPath(),
+                                    nCred.getPasswordAlias().getProviderPath()));
+                }
+            } else {
+                return false;
+            }
+        }
+        return false;
+    }
+
+    public static Credential getCredential(Datasource db) {
+        return getCredential(db, null);
+    }
+
+    public static Credential getCredential(Datasource db, Interfacetype interfaceType) {
+        if (interfaceType == null) {
+            return db.getInterfaces().getCredential();
+        } else {
+            for(Interface iface : db.getInterfaces().getInterfaces()) {
+                if (iface.getType() == interfaceType) {
+                    return iface.getCredential();
+                }
+            }
+        }
+        return null;
+    }
+
+    public static void validateCredential(Credential cred) throws FalconException {
+        if (cred == null) {
+            return;
+        }
+        switch (cred.getType()) {
+        case PASSWORD_TEXT:
+            if (StringUtils.isBlank(cred.getUserName()) || StringUtils.isBlank(cred.getPasswordText())) {
+                throw new FalconException(String.format("Credential type '%s' missing tags '%s' or '%s'",
+                    cred.getType().value(), "userName", "passwordText"));
+            }
+            break;
+        case PASSWORD_FILE:
+            if (StringUtils.isBlank(cred.getUserName()) || StringUtils.isBlank(cred.getPasswordFile())) {
+                throw new FalconException(String.format("Credential type '%s' missing tags '%s' or '%s'",
+                    cred.getType().value(), "userName", "passwordFile"));
+            }
+            break;
+        case PASSWORD_ALIAS:
+            if (StringUtils.isBlank(cred.getUserName()) || (cred.getPasswordAlias() == null)
+                || StringUtils.isBlank(cred.getPasswordAlias().getAlias())
+                || StringUtils.isBlank(cred.getPasswordAlias().getProviderPath())) {
+                throw new FalconException(String.format("Credential type '%s' missing tags '%s' or '%s' or %s'",
+                    cred.getType().value(), "userName", "alias", "providerPath"));
+            }
+            break;
+        default:
+            throw new FalconException(String.format("Unknown Credential type '%s'", cred.getType().value()));
+        }
+    }
+
+    private static boolean isSameString(String ep1, String ep2) {
```

Comment:
You can use existing function StringUtils::equals. The null case is already handled there.